calyx/src/analysis/live_range_analysis.rs

```diff
@@ -618,14 +656,29 @@ impl LiveRangeAnalysis {
         invoke: &ir::Invoke,
         shareable_components: &ShareSet,
     ) -> TypeNameSet {
-        invoke
+        let comb_group_uses: TypeNameSet =
+            if let Some(comb_group) = &invoke.comb_group {
+                ReadWriteSet::uses(comb_group.borrow().assignments.iter())
+                    .filter(|cell| {
+                        shareable_components.is_shareable_component(cell)
+                    })
+                    .map(|cell| {
+                        (cell.borrow().prototype.clone(), cell.clone_name())
+                    })
+                    .collect()
+            } else {
+                HashSet::new()
+            };
+        let mut invoke_uses: TypeNameSet = invoke
             .inputs
             .iter()
             .chain(invoke.outputs.iter())
             .filter_map(|(_, src)| {
                 Self::port_to_cell_name(src, shareable_components)
             })
-            .collect()
+            .collect();
+        invoke_uses.extend(comb_group_uses);
```

Comment:
I would reorder this so that you can call `extend` directly on the `comb_group_uses` iterator without using the `collect` call. If you move `invoke_uses` before the `if` you'll be able to do this